gst/nnstreamer/include/nnstreamer_plugin_api_trainer.h

```diff
@@ -0,0 +1,144 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * NNStreamer API for Tensor_Trainer Sub-Plugins
+ * Copyright (C) 2022 Hyunil Park <hyunil46.park@samsung.com>
+ */
+/**
+ * @file  nnstreamer_plugin_api_filter.h
+ * @date  1 Dec 2022
+ * @brief Mandatory APIs for NNStreamer Trainer sub-plugins (No External Dependencies)
+ * @see https://github.com/nnstreamer/nnstreamer
+ * @author  Hyunil Park <hyunil46.park@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+#ifndef __NNS_PLUGIN_API_TRAINER_H__
+#define __NNS_PLUGIN_API_TRAINER_H__
+
+#include "tensor_typedef.h"
+
+#define GST_TENSOR_TRAINER_FRAMEWORK_BASE (0xDEAFDEAD00000000ULL)
+#define GST_TENSOR_TRAINER_FRAMEWORK_V1 (GST_TENSOR_TRAINER_FRAMEWORK_BASE | 0x10000ULL)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief GstTensorTrainer's properties for neural network framework (internal data structure)
+ *
+ * Internal data of GstTensorTrainer required by tensor_trainer's custom subplugin.
+ */
+typedef struct _GstTensorTrainerProperties
+{
+  GstTensorsInfo input_meta;    /**< configured input tensor info */
+  const char *model_config;    /**< The configuration file path for creating model */
+  const char *model_save_path;    /**< The file path to save the model */
+  int64_t num_inputs;    /**< The number of input lists, the input is where framework receive the features to train the model, num_inputs indicates how many inputs there are. */
+  int64_t num_labels;    /**< The number of label lists, the label is where framework receive the class to train the model, num_labels indicates how many labels there are. */
+  int64_t num_train_samples;    /**< The number of train sample used to train the model. */
+  int64_t num_valid_samples;    /**< The number of valid sample used to train the model. */
+
+  GCond *train_complete_cond;    /**< Tensor trainer wait when receive EOS before model training is complete, subplugin should send signal when model train is complete */
+} GstTensorTrainerProperties;
+
+/**
+ * @brief Tensor_Trainer Subplugin framework related information
+ *
+ * All the information is provided statically.
+ */
+typedef struct _GstTensorTrainerFrameworkInfo
+{
+  const char *name;    /**< Name of the neural network framework, searchable by FRAMEWORK property. */
+} GstTensorTrainerFrameworkInfo;
+
+typedef struct _GstTensorTrainerFramework GstTensorTrainerFramework;
+
+/**
+ * @brief tensor_trainer Subplugin definition
+ *
+ * Common callback parameters:
+ * prop Trainer properties. Read Only.
+ * private_data Subplugin's private data. Set this (*private_data = XXX) if you want to change trainer->private_data.
+ */
+struct _GstTensorTrainerFramework
+{
+  uint64_t version;
```

Comment:
The format for the version looks the same as that of the tensor filter's version. Then, how about making it common, if possible?

Reply:
Thank you. Good idea.

Reply:
Taking this out and using it in common doesn't seem like something I can do right now.
I will do it with another PR when the policy and direction are decided.